### PR TITLE
Update TransAnnot version

### PR DESCRIPTION
**Pull request by mariia-zelenskaia**
Update the TransAnnot version for a new release

Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/transannot/meta.yaml [hunk 1]
```diff
@@ -1,12 +1,12 @@
-{% set version = "3-70b2a60" %}
-{% set sha256 = "3b1d2f28362004378528755b8b4e0b8dc45b51cc60562816258bdf16d9cda08d" %}
+{% set version = "3-e15e316" %}
```
Comment (bgruening) [inline recipes/transannot/meta.yaml:1]: @mariia-zelenskaia is this a hash in your version? Please note that if you do that there is no guarantee that one version is bigger than another. So you end up installing an old version by default.

**Comment by mariia-zelenskaia** [comment]
Why was the build number reset? Hash corresponds to the build number 3-
e15e316
<https://github.com/soedinglab/transannot/commit/e15e3168b68239f5cafa9b48b5d3ccbd6243a968>
.

Best,
Mariia

**Comment by mariia-zelenskaia** [comment]
It was reset after I've suggested changes

**Comment by martin-g** [comment]
Yes! Your first [commit](73e7fe9aa8df59511c1a26c1161ae350c4569f80) failed with a Lint error that the build number needs to be reset to 0. I did it in in the second commit and now the build is green.
But your version values do not increase, so `conda install ...` may install a random version.

**Comment by mariia-zelenskaia** [comment]
Thanks! So how do avoid conda installing a random version?

**Comment by martin-g** [comment]
> Thanks! So how do avoid conda installing a random version?

The best would be to use [Semantic Versioning](https://semver.org/).

Every time when you change the version you will have to reset the build number to 0.
If there is no change in the software but only in the recipe then keep the same version but increase the value of the build number with 1.

**Comment by mencian** [comment]
Closing in favor of #49875.

**Pull request closed without merging**